Makefile

```diff
@@ -80,67 +90,75 @@ bios/OVMF.fd: bios
 #
 run-installer-iso: bios/OVMF.fd
 	qemu-img create -f ${IMG_FORMAT} $(TARGET_IMG) ${MEDIA_SIZE}M
-	qemu-system-$(ZARCH) $(QEMU_OPTS) -drive file=$(TARGET_IMG),format=$(IMG_FORMAT) -cdrom $(INSTALLER_IMG).iso -boot d
+	$(QEMU_SYSTEM) $(QEMU_OPTS) -drive file=$(TARGET_IMG),format=$(IMG_FORMAT) -cdrom $(INSTALLER_IMG).iso -boot d
 
 run-installer-raw: bios/OVMF.fd
 	qemu-img create -f ${IMG_FORMAT} $(TARGET_IMG) ${MEDIA_SIZE}M
-	qemu-system-$(ZARCH) $(QEMU_OPTS) -drive file=$(TARGET_IMG),format=$(IMG_FORMAT) -drive file=$(INSTALLER_IMG).raw,format=raw
+	$(QEMU_SYSTEM) $(QEMU_OPTS) -drive file=$(TARGET_IMG),format=$(IMG_FORMAT) -drive file=$(INSTALLER_IMG).raw,format=raw
 
 run-fallback run: bios/OVMF.fd
-	qemu-system-$(ZARCH) $(QEMU_OPTS) -drive file=$(FALLBACK_IMG).img,format=$(IMG_FORMAT)
+	$(QEMU_SYSTEM) $(QEMU_OPTS) -drive file=$(FALLBACK_IMG).img,format=$(IMG_FORMAT)
 
 run-target: bios/OVMF.fd
-	qemu-system-$(ZARCH) $(QEMU_OPTS) -drive file=$(TARGET_IMG),format=$(IMG_FORMAT)
+	$(QEMU_SYSTEM) $(QEMU_OPTS) -drive file=$(TARGET_IMG),format=$(IMG_FORMAT)
 
 run-rootfs: bios/OVMF.fd bios/EFI
-	qemu-system-$(ZARCH) $(QEMU_OPTS) -drive file=$(ROOTFS_IMG),format=raw -drive file=fat:rw:./bios/,format=raw 
+	$(QEMU_SYSTEM) $(QEMU_OPTS) -drive file=$(ROOTFS_IMG),format=raw -drive file=fat:rw:./bios/,format=raw 
 
 run-grub: bios/OVMF.fd bios/EFI
-	qemu-system-$(ZARCH) $(QEMU_OPTS) -drive file=fat:rw:./bios/,format=raw
+	$(QEMU_SYSTEM) $(QEMU_OPTS) -drive file=fat:rw:./bios/,format=raw
+
+# ensure the dist directory exists
+$(DIST):
+	mkdir -p $(DIST)
+
+# convenience targets - so you can do `make config` instead of `make dist/config.img`, and `make installer` instead of `make dist/amd64/installer.img
+config: $(CONFIG_IMG)
+rootfs: $(ROOTFS_IMG)
+fallback: $(FALLBACK_IMG).img
+live: fallback
+installer: $(INSTALLER_IMG).raw
 
 # NOTE: that we have to depend on pkg/zedctr here to make sure
 # it gets triggered when we build any kind of image target
 images/%.yml: build-tools pkg/zedctr parse-pkgs.sh images/%.yml.in FORCE
 	$(PARSE_PKGS) $@.in > $@
-	@# the following is a horrible hack that needs to go away ASAP \
-	if [ "$(ZARCH)" = aarch64 ] ; then \
-           sed -e '/source:/s#rootfs.img#rootfs_aarch64.img#' -i.orig $@ ;\
-	   [ $$(uname -m) = aarch64 ] || echo "WARNING: We are assembling a $(ZARCH) image on `uname -m`. Things may break." ;\
-        fi
+	# the following is a horrible hack that needs to go away ASAP
+	@sed -e '/source:/s#ARCH#$(ARCH)#' -i.orig $@ 
```

Comment:
You sure this will work? Actually, a better question will be -- now that you introduced this nice structure why don't we get rid of this ugly hack completely by simply cd $(DIST) when we're building final images?

Reply:
> why don't we get rid of this ugly hack completely by simply cd $(DIST) when we're building final images?

Because... as I was doing this last night, I couldn't think of a cleaner way to do it. Now that you suggested it, we have one. I still prefer not `cd` if we can help it, but I think that it is a better way the the `sed` thing. This `sed` is better than what was before, but even so...

It would also mean having to move `dist/config.img` to `dist/$(ARCH)/config.img`, but I don't think that should matter all that much. There is nothing arch-specific about `config.img`, but it doesn't hurt, so why not?

Reply:
Working it now.